src/Compilers/CSharp/Portable/Utilities/TypeSymbolExtensions.cs

```diff
@@ -190,7 +190,7 @@ internal static TypeSymbol GetNextBaseTypeNoUseSiteDiagnostics(this TypeSymbol t
                     return GetNextDeclaredBase((NamedTypeSymbol)type, basesBeingResolved, compilation, ref visited);
 
                 default:
-                    // Enums and delegates know their own base types
+                    // Enums, delegates and extensions know their own base types
```

Comment:
>// Enums, delegates and extensions know their own base types [](http://example.com/codeflow?start=20&length=60)

Consider adding a PROTOTYPE comment instead. It looks like this method is used by lookup, which is not touched i this PR, and it is quite possible we would want to dig into the extended type here instead.